src/test/java/com/kentyou/featurelauncher/cli/FeatureLauncherCliTest.java

```diff
@@ -0,0 +1,501 @@
+		assertTrue(out.toString().contains("Using artifact repositories:"));
+		assertTrue(out.toString().contains(String.format("repositoryURI=%s", REMOTE_ARTIFACT_REPOSITORY_URI)));
+		assertTrue(out.toString().contains(String.format("name=%s", DEFAULT_REMOTE_ARTIFACT_REPOSITORY_NAME)));
+
+		assertEquals("", err.toString());
+	}
+
+	@Test
+	public void testArtifactRepositoryOptionLong() {
+		StringBuilder args = new StringBuilder();
+		args.append("--dry-run");
+		args.append(" ");
+		args.append("--artifact-repository=");
+		args.append(REMOTE_ARTIFACT_REPOSITORY_URI);
+		args.append(" ");
+		args.append("--artifact-repository=");
+		args.append(REMOTE_ARTIFACT_REPOSITORY_NAME);
+		args.append("=");
+		args.append(DEFAULT_REMOTE_ARTIFACT_REPOSITORY_NAME);
+		args.append(" ");
+		args.append(FEATURE_FILE_PATH);
+
+		int exitCode = new CommandLine(new FeatureLauncherCli()).execute((args.toString().split(" ")));
+		assertEquals(0, exitCode);
+
+		assertTrue(out.toString().contains(String.format("Launching feature %s", FEATURE_FILE_ID)));
+		assertTrue(out.toString().contains("Using artifact repositories:"));
+		assertTrue(out.toString().contains(String.format("repositoryURI=%s", REMOTE_ARTIFACT_REPOSITORY_URI)));
+		assertTrue(out.toString().contains(String.format("name=%s", DEFAULT_REMOTE_ARTIFACT_REPOSITORY_NAME)));
+
+		assertEquals("", err.toString());
+	}
+
+	@Test
+	public void testDecoratorOptionShort() {
+		StringBuilder args = new StringBuilder();
+		args.append("--dry-run");
+		args.append(" ");
+		args.append("-d=");
+		args.append(DECORATOR_CLASS_NAME);
+		args.append(" ");
+		args.append(FEATURE_FILE_PATH);
+
+		int exitCode = new CommandLine(new FeatureLauncherCli()).execute((args.toString().split(" ")));
+		assertEquals(0, exitCode);
+
+		assertTrue(out.toString().contains(String.format("Launching feature %s", FEATURE_FILE_ID)));
+		assertTrue(out.toString().contains("Using decorators:"));
+		assertTrue(out.toString().contains(String.format("class %s", DECORATOR_CLASS_NAME)));
+
+		assertEquals("", err.toString());
+	}
+
+	@Test
+	public void testDecoratorOptionLong() {
+		StringBuilder args = new StringBuilder();
+		args.append("--dry-run");
+		args.append(" ");
+		args.append("--decorator=");
+		args.append(DECORATOR_CLASS_NAME);
+		args.append(" ");
+		args.append(FEATURE_FILE_PATH);
+
+		int exitCode = new CommandLine(new FeatureLauncherCli()).execute((args.toString().split(" ")));
+		assertEquals(0, exitCode);
+
+		assertTrue(out.toString().contains(String.format("Launching feature %s", FEATURE_FILE_ID)));
+		assertTrue(out.toString().contains("Using decorators:"));
+		assertTrue(out.toString().contains(String.format("class %s", DECORATOR_CLASS_NAME)));
+
+		assertEquals("", err.toString());
+	}
+
+	@Test
+	public void testExtensionHandlerOptionShort() {
+		StringBuilder args = new StringBuilder();
+		args.append("--dry-run");
+		args.append(" ");
+		args.append("-e=");
+		args.append(EXTENSION_HANDLER_NAME);
```

Comment:
The specification chapter uses `-e extension.name,class.name` with a `,` not an `=`. Would you like me to update the spec text as this probably does read better?

Reply:
I think if we're talking about key value pairs, `key=value` syntax is better suited than `key,value`, so in this particular case I think specification would need to be updated.